Theia v1.61.0 npm+package.json build failure: webpack exited with an unexpected code: 1

<!-- Please provide a detailed description of the bug. -->
<!-- Note: This template is not meant for security vulnerabilities disclosure -->
<!-- Any such issue, created in this repo, will be deleted on sight -->
<!-- Instead please report vulnerabilities to the Eclipse Foundation's security team -->
<!-- For more details, please read SECURITY.md in the repository root -->
### Bug Description:

Building Theia v1.61.0 browser application using near-standard package.json fails with "webpack exited with an unexpected code: 1", and "Module not found: Error: Can't resolve 'lodash/debounce'" despite "webpack 5.99.7 compiled successfully" message.

Failure summary:

```
#12 50.62 ERROR in ./node_modules/@theia/notebook/lib/browser/notebook-editor-widget.js 37:17-43
#12 50.62 Module not found: Error: Can't resolve 'lodash/debounce' in '/theia/node_modules/@theia/notebook/lib/browser'
#12 50.62 resolve 'lodash/debounce' in '/theia/node_modules/@theia/notebook/lib/browser'

[SNIP]

#12 49.14 webpack 5.99.7 compiled successfully in 26199 ms
#12 49.14 
#12 49.47 Error: webpack exited with an unexpected code: 1.
#12 49.47     at ChildProcess.<anonymous> (/theia/node_modules/@theia/application-manager/lib/application-process.js:86:28)
#12 49.47     at ChildProcess.emit (node:events:524:28)
#12 49.47     at maybeClose (node:internal/child_process:1104:16)
#12 49.47     at ChildProcess._handle.onexit (node:internal/child_process:304:5)
#12 ERROR: process "/bin/bash -c npm run bundle" did not complete successfully: exit code: 1
```

<!-- Please provide clear steps to reproduce the bug. -->
### Steps to Reproduce:

Remove `@theia/test`, `@theia/api-samples` and `@theia/api-provider-sample` from v1.61.0's `examples/browser/package.json` and run `npm install` then `npm run bundle`:

```
mkdir -p /tmp/theia && cd /tmp/theia && cat <<_EOE_ >>Dockerfile && docker build -t theia-build --progress=plain .
# syntax=docker/dockerfile:1.3-labs

FROM node:20 AS theia-build-environment

SHELL ["/bin/bash", "-c"]

RUN apt-get update && \
    apt-get -y install make gcc pkg-config libx11-dev libxkbfile-dev libsecret-1-dev python3 curl

RUN mkdir /theia && \
    curl https://raw.githubusercontent.com/eclipse-theia/theia/refs/tags/v1.61.0/examples/browser/package.json | egrep -v '"@theia/(api-|test)' >/theia/package.json

ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=1
ENV PUPPETEER_SKIP_DOWNLOAD=1

WORKDIR /theia

RUN npm install

ENV NODE_OPTIONS="--max-old-space-size=8192"
RUN npm run bundle
RUN npm run bundle:production
_EOE_
```

The resulting logs are:

```
#0 building with "default" instance using docker driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 648B done
#1 DONE 0.0s

#2 [auth] docker/dockerfile:pull token for registry-1.docker.io
#2 DONE 0.0s

#3 resolve image config for docker-image://docker.io/docker/dockerfile:1.3-labs
#3 DONE 0.7s

#4 docker-image://docker.io/docker/dockerfile:1.3-labs@sha256:250ce669e1aeeb5ffb892b18039c3f0801466536cb4210c8eb2638e628859bfd
#4 CACHED

#5 [internal] load .dockerignore
#5 transferring context: 2B done
#5 DONE 0.0s

#6 [internal] load metadata for docker.io/library/node:20
#6 DONE 0.0s

#7 [1/7] FROM docker.io/library/node:20
#7 DONE 0.0s

#8 [2/7] RUN apt-get update &&     apt-get -y install make gcc pkg-config libx11-dev libxkbfile-dev libsecret-1-dev python3 curl
#8 CACHED

#9 [3/7] RUN mkdir /theia &&     curl https://raw.githubusercontent.com/eclipse-theia/theia/refs/tags/v1.61.0/examples/browser/package.json | egrep -v '"@theia/(api-|test)' >/theia/package.json
#9 CACHED

#10 [4/7] WORKDIR /theia
#10 CACHED

#11 [5/7] RUN npm install
#11 CACHED

#12 [6/7] RUN npm run bundle
#12 0.592 
#12 0.592 > @theia/example-browser@1.61.0 bundle
#12 0.592 > npm run rebuild && theia build --mode development
#12 0.592 
#12 0.903 
#12 0.903 > @theia/example-browser@1.61.0 rebuild
#12 0.903 > theia rebuild:browser --cacheRoot ../..
#12 0.903 
#12 1.629 native node modules are already rebuilt for browser
#12 2.281 Could not resolve optional peer dependency '@theia/electron'. Skipping...
#12 50.62 assets by status 1.92 MiB [cached] 14 assets
#12 50.62 assets by chunk 18.3 MiB (id hint: vendors)
#12 50.62   asset vendors-node_modules_theia_core_lib_common_markdown-rendering_index_js-node_modules_theia_plu-b1545a.js 4.55 MiB [emitted] (id hint: vendors) 2 related assets
#12 50.62   asset vendors-node_modules_theia_preview_lib_browser_preview-frontend-module_js.js 1.4 MiB [emitted] (id hint: vendors) 2 related assets
#12 50.62   asset vendors-node_modules_theia_ai-chat-ui_lib_browser_ai-chat-ui-frontend-module_js.js 1.4 MiB [emitted] (id hint: vendors) 2 related assets
#12 50.62   asset vendors-node_modules_theia_collaboration_lib_browser_collaboration-frontend-module_js.js 778 KiB [emitted] (id hint: vendors) 2 related assets
#12 50.62   + 101 assets
#12 50.62 assets by path ../webview/pre/ 40.5 KiB
#12 50.62   assets by path ../webview/pre/*.js 39.8 KiB
#12 50.62     asset ../webview/pre/main.js 24.9 KiB [emitted] [from: node_modules/@theia/plugin-ext/src/main/browser/webview/pre/main.js] [copied] 1 related asset
#12 50.62     + 2 assets
#12 50.62   assets by path ../webview/pre/*.html 726 bytes
#12 50.62     asset ../webview/pre/index.html 480 bytes [emitted] [from: node_modules/@theia/plugin-ext/src/main/browser/webview/pre/index.html] [copied] 1 related asset
#12 50.62     asset ../webview/pre/fake.html 246 bytes [emitted] [from: node_modules/@theia/plugin-ext/src/main/browser/webview/pre/fake.html] [copied]
#12 50.62 + 19 assets
#12 50.62 4576 modules
#12 50.62   
#12 50.62 
#12 50.62 WARNING in ./node_modules/@theia/monaco-editor-core/esm/vs/base/common/worker/simpleWorker.js 439:56-66
#12 50.62 Critical dependency: the request of a dependency is an expression
#12 50.62  @ ./node_modules/@theia/monaco-editor-core/esm/vs/editor/common/services/editorWorkerBootstrap.js 9:26-80
#12 50.62  @ ./node_modules/@theia/monaco-editor-core/esm/vs/editor/editor.worker.js 21:13-66
#12 50.62 
#12 50.62 WARNING in ./node_modules/@theia/monaco-editor-core/esm/vs/editor/common/services/editorSimpleWorker.js 420:53-63
#12 50.62 Critical dependency: the request of a dependency is an expression
#12 50.62  @ ./node_modules/@theia/monaco-editor-core/esm/vs/editor/common/services/editorWorkerBootstrap.js 10:32-66
#12 50.62  @ ./node_modules/@theia/monaco-editor-core/esm/vs/editor/editor.worker.js 21:13-66
#12 50.62 
#12 50.62 2 warnings have detailed information that is not shown.
#12 50.62 Use 'stats.errorDetails: true' resp. '--stats-error-details' to show it.
#12 50.62 
#12 50.62 ERROR in ./node_modules/@theia/notebook/lib/browser/notebook-editor-widget.js 37:17-43
#12 50.62 Module not found: Error: Can't resolve 'lodash/debounce' in '/theia/node_modules/@theia/notebook/lib/browser'
#12 50.62 resolve 'lodash/debounce' in '/theia/node_modules/@theia/notebook/lib/browser'
#12 50.62   Parsed request is a module
#12 50.62   using description file: /theia/node_modules/@theia/notebook/package.json (relative path: ./lib/browser)
#12 50.62     Field 'browser' doesn't contain a valid alias configuration
#12 50.62     resolve as module
#12 50.62       /theia/node_modules/@theia/notebook/lib/browser/node_modules doesn't exist or is not a directory
#12 50.62       /theia/node_modules/@theia/notebook/lib/node_modules doesn't exist or is not a directory
#12 50.62       /theia/node_modules/@theia/notebook/node_modules doesn't exist or is not a directory
#12 50.62       /theia/node_modules/@theia/node_modules doesn't exist or is not a directory
#12 50.62       /theia/node_modules/node_modules doesn't exist or is not a directory
#12 50.62       looking for modules in /theia/node_modules
#12 50.62         /theia/node_modules/lodash doesn't exist
#12 50.62       /node_modules doesn't exist or is not a directory
#12 50.62  @ ./node_modules/@theia/notebook/lib/browser/notebook-frontend-module.js 33:33-68
#12 50.62  @ ./src-gen/frontend/index.js 132:30-92
#12 50.62 
#12 50.62 ERROR in ./node_modules/@theia/notebook/lib/browser/view/notebook-find-widget.js 22:17-43
#12 50.62 Module not found: Error: Can't resolve 'lodash/debounce' in '/theia/node_modules/@theia/notebook/lib/browser/view'
#12 50.62 resolve 'lodash/debounce' in '/theia/node_modules/@theia/notebook/lib/browser/view'
#12 50.62   Parsed request is a module
#12 50.62   using description file: /theia/node_modules/@theia/notebook/package.json (relative path: ./lib/browser/view)
#12 50.62     Field 'browser' doesn't contain a valid alias configuration
#12 50.62     resolve as module
#12 50.62       /theia/node_modules/@theia/notebook/lib/browser/view/node_modules doesn't exist or is not a directory
#12 50.62       /theia/node_modules/@theia/notebook/lib/browser/node_modules doesn't exist or is not a directory
#12 50.62       /theia/node_modules/@theia/notebook/lib/node_modules doesn't exist or is not a directory
#12 50.62       /theia/node_modules/@theia/notebook/node_modules doesn't exist or is not a directory
#12 50.62       /theia/node_modules/@theia/node_modules doesn't exist or is not a directory
#12 50.62       /theia/node_modules/node_modules doesn't exist or is not a directory
#12 50.62       looking for modules in /theia/node_modules
#12 50.62         /theia/node_modules/lodash doesn't exist
#12 50.62       /node_modules doesn't exist or is not a directory
#12 50.62  @ ./node_modules/@theia/notebook/lib/browser/notebook-editor-widget.js 36:31-69
#12 50.62  @ ./node_modules/@theia/notebook/lib/browser/notebook-frontend-module.js 33:33-68
#12 50.62  @ ./src-gen/frontend/index.js 132:30-92
#12 50.62 
#12 50.62 webpack 5.99.7 compiled with 2 errors and 2 warnings in 45763 ms
#12 50.62 
#12 50.62 assets by status 237 KiB [cached] 2 assets
#12 50.62 assets by info 1.69 MiB [immutable]
#12 50.62   assets by path *.woff2 256 KiB
#12 50.62     asset 2f64b833219dc8433424.woff2 88.6 KiB [emitted] [immutable] [from: node_modules/file-icons-js/fonts/file-icons.woff2] (auxiliary name: secondary-window)
#12 50.62     + 4 assets
#12 50.62   assets by path *.ttf 321 KiB
#12 50.62     asset 1e59d2330b4c6deb84b3.ttf?v=4.7.0 162 KiB [emitted] [immutable] [from: node_modules/font-awesome/fonts/fontawesome-webfont.ttf?v=4.7.0] (auxiliary name: secondary-window)
#12 50.62     + 2 assets
#12 50.62   asset 55a4dd893aa466108ee8.wasm 462 KiB [emitted] [immutable] [from: node_modules/vscode-oniguruma/release/onig.wasm] (auxiliary name: secondary-window)
#12 50.62   asset c1e38fd9e0e74ba58f7a.svg?v=4.7.0 434 KiB [emitted] [immutable] [from: node_modules/font-awesome/fonts/fontawesome-webfont.svg?v=4.7.0] (auxiliary name: secondary-window)
#12 50.62   asset 8b43027f47b20503057d.eot?v=4.7.0 162 KiB [emitted] [immutable] [from: node_modules/font-awesome/fonts/fontawesome-webfont.eot?v=4.7.0] (auxiliary name: secondary-window)
#12 50.62   asset f691f37e57f04c152e23.woff?v=4.7.0 95.7 KiB [emitted] [immutable] [from: node_modules/font-awesome/fonts/fontawesome-webfont.woff?v=4.7.0] (auxiliary name: secondary-window)
#12 50.62 assets by chunk 25.5 MiB (name: secondary-window)
#12 50.62   asset secondary-window.js 25 MiB [emitted] (name: secondary-window) 1 related asset
#12 50.62   asset secondary-window.css 529 KiB [emitted] (name: secondary-window) 1 related asset
#12 50.62 Entrypoint secondary-window 25.5 MiB (27.6 MiB) = secondary-window.css 529 KiB secondary-window.js 25 MiB 16 auxiliary assets
#12 50.62 runtime modules 105 KiB 463 modules
#12 50.62 orphan modules 1.22 MiB (javascript) 1.47 MiB (asset) [orphan] 172 modules
#12 50.62 javascript modules 20.8 MiB 2937 modules
#12 50.62 css modules 470 KiB 146 modules
#12 50.62 json modules 1.74 MiB
#12 50.62   modules by path ./node_modules/@theia/core/ 1.58 MiB 50 modules
#12 50.62   modules by path ./node_modules/@theia/monaco/ 34.5 KiB 10 modules
#12 50.62   modules by path ./node_modules/iconv-lite/ 86.7 KiB 8 modules
#12 50.62   modules by path ./node_modules/ajv/lib/ 3.12 KiB 2 modules
#12 50.62   ./node_modules/@theia/ai-core/data/prompttemplate.tmLanguage.json 1.5 KiB [built] [code generated]
#12 50.62   ./node_modules/entities/lib/maps/entities.json 28.4 KiB [built] [code generated]
#12 50.62 ./node_modules/vscode-oniguruma/release/onig.wasm 462 KiB (asset) 42 bytes (javascript) [built] [code generated]
#12 50.62   
#12 50.62 
#12 50.62 WARNING in ./node_modules/@theia/monaco-editor-core/esm/vs/base/common/worker/simpleWorker.js 439:56-66
#12 50.62 Critical dependency: the request of a dependency is an expression
#12 50.62  @ ./node_modules/@theia/monaco-editor-core/esm/vs/editor/browser/services/editorWorkerService.js 19:26-80
#12 50.62  @ ./node_modules/@theia/monaco-editor-core/esm/vs/editor/standalone/browser/standaloneServices.js 72:33-89
#12 50.62  @ ./node_modules/@theia/debug/lib/browser/debug-frontend-module.js 54:29-117
#12 50.62  @ ./src-gen/frontend/secondary-index.js 13:19-84
#12 50.62 
#12 50.62 WARNING in ./node_modules/@theia/monaco-editor-core/esm/vs/editor/common/services/editorSimpleWorker.js 420:53-63
#12 50.62 Critical dependency: the request of a dependency is an expression
#12 50.62  @ ./node_modules/@theia/monaco-editor-core/esm/vs/editor/browser/services/editorWorkerService.js 23:32-86
#12 50.62  @ ./node_modules/@theia/monaco-editor-core/esm/vs/editor/standalone/browser/standaloneServices.js 72:33-89
#12 50.62  @ ./node_modules/@theia/debug/lib/browser/debug-frontend-module.js 54:29-117
#12 50.62  @ ./src-gen/frontend/secondary-index.js 13:19-84
#12 50.62 
#12 50.62 2 warnings have detailed information that is not shown.
#12 50.62 Use 'stats.errorDetails: true' resp. '--stats-error-details' to show it.
#12 50.62 
#12 50.62 webpack 5.99.7 compiled with 2 warnings in 21000 ms
#12 50.62 
#12 50.62 assets by path *.js 18.3 MiB
#12 50.62   assets by chunk 18.2 MiB (id hint: vendors)
#12 50.62     asset vendors-node_modules_drivelist_build_Release_drivelist_node-node_modules_stroncium_procfs_lib-deaf8b.js 11.4 MiB [emitted] (id hint: vendors) 1 related asset
#12 50.62     asset vendors-node_modules_theia_core_lib_common_index_js.js 2.31 MiB [emitted] (id hint: vendors) 1 related asset
#12 50.62     asset vendors-node_modules_theia_core_shared_reflect-metadata_index_js-node_modules_theia_plugin-ex-2556b2.js 1.89 MiB [emitted] (id hint: vendors) 1 related asset
#12 50.62     asset vendors-node_modules_theia_core_lib_common_collections_js-node_modules_theia_core_lib_common_-b5530b.js 1.49 MiB [emitted] (id hint: vendors) 1 related asset
#12 50.62     + 7 assets
#12 50.62   + 8 assets
#12 50.62 assets by path native/*.node 715 KiB
#12 50.62   asset native/watcher.node 503 KiB [emitted] (auxiliary id hint: vendors)
#12 50.62   asset native/keytar.node 75.2 KiB [emitted] (auxiliary id hint: vendors)
#12 50.62   asset native/pty.node 69.9 KiB [emitted] (auxiliary id hint: vendors)
#12 50.62   asset native/drivelist.node 66.9 KiB [emitted] (auxiliary id hint: vendors)
#12 50.62 asset worker/conoutSocketWorker.js 3.4 KiB [emitted] (name: worker/conoutSocketWorker) 1 related asset
#12 50.62 runtime modules 27.4 KiB 71 modules
#12 50.62 cacheable modules 16.5 MiB
#12 50.62   javascript modules 14.1 MiB
#12 50.62     modules by path ./node_modules/ 14.1 MiB 2325 modules
#12 50.62     modules by path ./src-gen/backend/*.js 10.2 KiB 2 modules
#12 50.62     modules by path ./lib/backend/native-webpack-plugin/*.js 361 bytes 2 modules
#12 50.62   json modules 2.42 MiB 48 modules
#12 50.62 + 56 modules
#12 50.62 webpack 5.99.7 compiled successfully in 27618 ms
#12 50.62 
#12 50.95 Error: webpack exited with an unexpected code: 1.
#12 50.95     at ChildProcess.<anonymous> (/theia/node_modules/@theia/application-manager/lib/application-process.js:86:28)
#12 50.95     at ChildProcess.emit (node:events:518:28)
#12 50.95     at maybeClose (node:internal/child_process:1105:16)
#12 50.95     at ChildProcess._handle.onexit (node:internal/child_process:305:5)
#12 ERROR: process "/bin/bash -c npm run bundle" did not complete successfully: exit code: 1
------
 > [6/7] RUN npm run bundle:
50.62     modules by path ./lib/backend/native-webpack-plugin/*.js 361 bytes 2 modules
50.62   json modules 2.42 MiB 48 modules
50.62 + 56 modules
50.62 webpack 5.99.7 compiled successfully in 27618 ms
50.62 
50.95 Error: webpack exited with an unexpected code: 1.
50.95     at ChildProcess.<anonymous> (/theia/node_modules/@theia/application-manager/lib/application-process.js:86:28)
50.95     at ChildProcess.emit (node:events:518:28)
50.95     at maybeClose (node:internal/child_process:1105:16)
50.95     at ChildProcess._handle.onexit (node:internal/child_process:305:5)
------
Dockerfile:19
--------------------
  17 |     
  18 |     ENV NODE_OPTIONS="--max-old-space-size=8192"
  19 | >>> RUN npm run bundle
  20 |     RUN npm run bundle:production
  21 |     
--------------------
ERROR: failed to solve: process "/bin/bash -c npm run bundle" did not complete successfully: exit code: 1
```


<!-- Please provide any additional information available. -->
<!-- Additional information can be in the form of logs, screenshots, screencasts. -->

### Additional Information

- Operating System: Debian Bookworm with Node v20
- Theia Version: v1.61.0
- Architecture: Reproduced on amd64 and arm64

--- a/packages/notebook/src/browser/view/notebook-find-widget.tsx
+++ b/packages/notebook/src/browser/view/notebook-find-widget.tsx
@@ -17,7 +17,7 @@
 import { nls } from '@theia/core';
 import * as React from '@theia/core/shared/react';
 import { codicon } from '@theia/core/lib/browser';
-import debounce = require('lodash/debounce');
+import debounce = require('@theia/core/shared/lodash.debounce');
 
 export interface NotebookEditorFindMatch {
     selected: boolean;

